### PR TITLE
Manifest enhancements

### PR DESCRIPTION
**Pull request by Kubuxu**
- Save latest manifest to datastore
- Add protocol version to manifest

### CODE REVIEW

**Comment by Kubuxu** [comment]
The power table is still WIP

File: f3.go [hunk 1]
```diff
@@ -242,14 +242,19 @@ func (m *F3) reconfigure(ctx context.Context, manifest *manifest.Manifest) (_err
...
 	}
 
-	if manifest == nil {
+	// pause if new manifest is nil
+	if manif == nil {
+		return nil
+	}
+	// pause if we are not capable of supporting this version
+	if manif.ProtocolVersion > manifest.VersionCapability {
 		return nil
```
Comment (Stebalien) [inline f3.go:251]: I'd log something (once).

File: manifest/dynamic_manifest.go [hunk 2]
```diff
@@ -98,13 +105,27 @@ func (m *DynamicManifestProvider) Start(startCtx context.Context) (_err error) {
...
+		msgSeqNumber = update.MessageSequence
+		currentManifest = update.Manifest
+	}
 
-	currentManifest := m.initialManifest
-	m.manifestChanges <- m.initialManifest
+	m.manifestChanges <- currentManifest
 
-	m.errgrp.Go(func() error {
+	m.errgrp.Go(func() (_err error) {
```
Comment (Stebalien) [inline manifest/dynamic_manifest.go:128]: Oops...

File: f3.go (hunk 1 above)
Comment (Stebalien) [reply f3.go:251]: Also, maybe we should have an equality here?

**Review by Stebalien — APPROVED** [review]

File: f3.go (hunk 1 above)
Comment (Kubuxu) [reply f3.go:251]: this IMO gives us more flexibility

Comment (Stebalien) [reply f3.go:251]: Eh, up to you. I guess we're unlikely to really go "backwards" anyways.

**Pull request merged**